Sources/SwiftSyntax/SyntaxRewriter.swift.gyb

```diff
@@ -58,44 +74,130 @@ open class SyntaxRewriter {
   ///   - node: the node we just finished visiting.
   open func visitPost(_ node: Syntax) {}
 
-  public func visit(_ node: Syntax) -> Syntax {
-    visitPre(node)
-    defer { visitPost(node) }
-
-    // If the global visitor returned non-nil, skip specialized dispatch.
-    if let newNode = visitAny(node) {
-      return newNode
-    }
+  /// Visit any Syntax node. 
+  ///   - Parameter node: the node that is being visited
+  ///   - Returns: the rewritten node
+  public func visit(_ node: Syntax) -> Syntax? {
```

Comment:
Is the return type being optional here intentional?

Reply:
No. Thanks for catching this. Must have been a leftover from when I tried not returning the Syntax node.